examples/extra_element_integer/random_cluster.cpp

```diff
@@ -0,0 +1,154 @@
+#include "libmesh/dof_map.h"
+#include "libmesh/elem.h"
+#include "libmesh/equation_systems.h"
+#include "libmesh/exodusII_io.h"
+#include "libmesh/libmesh.h"
+#include "libmesh/linear_implicit_system.h"
+#include "libmesh/mesh.h"
+#include "libmesh/mesh_generation.h"
+#include "libmesh/numeric_vector.h"
+#include "libmesh/system.h"
+
+#include <cstdlib>
+#include <ctime>
+#include <random>
+#include <vector>
+
+using namespace libMesh;
+
+void CreateMesh(libMesh::Mesh &mesh, int nx, int ny) {
+  libMesh::MeshTools::Generation::build_square(mesh, nx, ny, 0.0, 10.0, 0.0,
+                                               10.0);
+}
+
+void CreateMesh(libMesh::Mesh &mesh, int nx, int ny, int nz) {
+  libMesh::MeshTools::Generation::build_cube(mesh, nx, ny, nz, 0.0, 10.0, 0.0,
+                                             10.0, 0.0, 10.0);
+}
+
+void PopulateRandomIntegers(libMesh::Mesh &mesh, LinearImplicitSystem &system) {
+  DofMap &dof_map = system.get_dof_map();
+  for (const auto &elem : mesh.element_ptr_range()) {
+
+    std::vector<dof_id_type> dof_indices;
+    dof_map.dof_indices(elem, dof_indices);
+    int random_int = rand() % 5;
+    system.solution->set(dof_indices[0], 5 - random_int);
+  }
+  system.solution->close();
+
+}
+
+void AddVariablesToSystem(libMesh::EquationSystems &equation_systems,
+                          LinearImplicitSystem &system,
+                          std::string var_name, bool needs_re_init =false) {
+    system.add_variable(var_name, CONSTANT, MONOMIAL);
+    if (!needs_re_init){
+        equation_systems.init();
+    }
+    else{
+        equation_systems.reinit();
+    }
+}
+
+const unsigned int FindCluster(libMesh::Mesh &mesh, LinearImplicitSystem &system,
+                 const std::string & variable_name) {
+
+  const unsigned int variable_num = system.variable_number(variable_name);
+  DofMap &dof_map = system.get_dof_map();
+  const unsigned int index = mesh.add_elem_integer(variable_name);
+
+  for (const auto &elem : mesh.element_ptr_range()) {
+
+    std::vector<dof_id_type> global_dof_indices;
+    dof_map.dof_indices(elem, global_dof_indices, variable_num);
+
+    std::vector<double> solution_value(1);
+    system.solution->get(global_dof_indices, solution_value);
+    int element_solution = static_cast<int>(solution_value[0]);
```

Comment:
This looks like a good function that we can reuse on the neighbor below.

(Related: it takes a lot of code to get a field variable from a mesh! 😀)